backend/app/services/search.py

```diff
@@ -1,42 +1,222 @@
 from elasticsearch import AsyncElasticsearch
 from fastapi.encoders import jsonable_encoder
-
+from datetime import datetime, timedelta
 from app import config
+import bisect
 
-
+    
 class ElasticService:
     # todo add bulkhead pattern
     # todo add error message for unauthorized user
 
-    def __init__(self,configpath="",index=""):
-        self.cfg = config.get_config()
-        self.url = self.cfg.get(configpath+'.url')
-        self.esUser = None
+    def initialize_es(self, config, path, index):
+        url = config.get(path+'.url')
+        esUser = None
         if index == "":
-            self.indice = self.cfg.get(configpath+'.indice')
+            indice = config.get(path+'.indice')
         else:
-            self.indice = index
-        if self.cfg.is_set(configpath+'.username') and \
-                self.cfg.is_set(configpath+'.password'):
-            self.esUser = self.cfg.get(configpath+'.username')
-            self.esPass = self.cfg.get(configpath+'.password')
-        if self.esUser :
-            self.es = AsyncElasticsearch(
-                    self.url,
+            indice = index
+        if config.is_set(path+'.username') and \
+                config.is_set(path+'.password'):
+            esUser = config.get(path+'.username')
+            esPass = config.get(path+'.password')
+        if esUser :
+            es = AsyncElasticsearch(
+                    url,
                     use_ssl=False,
                     verify_certs=False,
-                    http_auth=(self.esUser,self.esPass)
+                    http_auth=(esUser,esPass)
             )
         else:
-            self.es = AsyncElasticsearch(self.url, verify_certs=False)
+            es = AsyncElasticsearch(url, verify_certs=False)
+        return es, indice
+
+    def __init__(self,configpath="",index=""):
```

Comment:
Can you move __init__ to the top of the class?

Reply:
Organised as suggested.